include/gzip/decompress.hpp

```diff
@@ -47,7 +50,7 @@ std::string decompress(const char* data, std::size_t size)
     {
         output.resize(length + 2 * size);
         inflate_s.avail_out = static_cast<unsigned int>(2 * size);
-        inflate_s.next_out = (Bytef*)(output.data() + length);
+        inflate_s.next_out = reinterpret_cast<Bytef*>(&output[0] + length);
```

Comment:
`output.data()` returns a const value. But in this case `output.data()` is the pointer we write to - so we need the non-const/mutable pointer here.